libsolidity/formal/CHC.cpp

```diff
@@ -1584,6 +1638,32 @@ optional<string> CHC::generateCounterexample(CHCSolverInterface::CexGraph const&
 				auto const& outParams = calledFun->returnParameters();
 				if (auto outStr = formatVariableModel(outParams, outValues, "\n"); !outStr.empty())
 					localState += outStr + "\n";
+
+				optional<unsigned> localErrorId;
+				solidity::util::BreadthFirstSearch<unsigned> bfs{{summaryId}};
+				bfs.run([&](auto _nodeId, auto&& _addChild) {
+					auto const& edges = _graph.edges.at(_nodeId);
+					if (edges.size() == 1)
+					{
+						auto const* child = nodePred(edges.front());
+						if (child->isFunctionErrorBlock())
+						{
+							localErrorId = edges.front();
+							bfs.abort();
+						}
+					}
+					for (auto c: edges)
+						_addChild(c);
```

Comment:
Is it better using something like `std::for_each` in such cases?

Reply:
I don't like `std::for_each` because it takes iterators instead of the container. But I replaced it by `ranges::for_each` which takes the container